catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/SegmentHandler.java

```diff
@@ -32,6 +37,19 @@
                 .forEach(attribute -> handleValue(metacard, attribute, segment));
     }
 
+    protected void handleTres(Metacard metacard,
+            TaggedRecordExtensionHandler taggedRecordextensionHandler) {
+        List<Tre> tres = taggedRecordextensionHandler.getTREsRawStructure()
+                .getTREs();
+        tres.stream()
+                .forEach(tre -> Arrays.stream(TreDescriptor.values())
+                        .filter(treDescriptor -> tre.getName()
+                                .trim()
+                                .equals(treDescriptor.name()))
+                        .findFirst()
+                        .ifPresent(first -> handleSegmentHeader(metacard, tre, first.getValues())));
+    }
+
```

Comment:
I think this can be simplified.  https://gist.github.com/dcruver/454db70c8052d03f6da19dce83f83a56